boot.c

```diff
@@ -200,6 +208,12 @@ eyrie_boot(uintptr_t dummy, // $a0 contains the return value from the SBI
 
   /* initialize free memory */
   init_freemem();
+
+  /* initialize user stack */
+  init_user_stack();
+
+  /* settup user stack env/aux */
+  setup_start((void*)runtime_va_start);
```

Comment:
setup_start takes the current user stack pointer, and then returns a new user stack pointer (offset from the previous) to give to programs that use libc init. need to update this